specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2964 @@
+          "readOnly": true
+        },
+        "clientId": {
+          "description": "Client Id of user assigned identity",
+          "type": "string",
+          "x-ms-client-name": "ClientId",
+          "readOnly": true
+        }
+      },
+      "description": "Metadata of user assigned identity."
+    },
+    "CutoverMigrationRequest": {
+      "type": "object",
+      "description": "Cutover migration request",
+      "properties": {
+        "dnsChangeMode": {
+          "type": "string",
+          "enum": [
+            "Update",
+            "Revert",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DnsChangeMode",
+            "modelAsString": true
+          },
+          "description": "The DNS Mode"
+        }
+      }
+    },
+    "MigrationAssessmentRequest": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        }
+      },
+      "description": "Request from client to perform migration assessment"
+    },
+    "MigrationAssessmentResource": {
+      "type": "object",
+      "properties": {
+        "sourceServerResourceId": {
+          "type": "string",
+          "description": "Source Server Resource Id to verify and perform migration assessment on."
+        },
+        "targetPlatformProviderId": {
+          "type": "string",
+          "description": "Target platform(Flexible Server) provider ID"
+        },
+        "targetServerDetails": {
+          "$ref": "#/definitions/TargetServerDetails"
+        },
+        "summary": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AssessmentSummaryItem"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "featureName"
+          ],
+          "description": "Represents a summary of Migration Assessment"
+        },
+        "replicaAssessmentReports": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MigrationAssessmentResource"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "sourceServerResourceId"
+          ],
+          "description": "Represents Migration assessment resource of replica servers"
+        },
+        "assessmentReportCompletionDateTime": {
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true,
+          "description": "Assessment report completion date time"
+        }
+      },
+      "description": "Represents a resource name availability."
+    },
+    "TargetServerDetails": {
+      "type": "object",
+      "properties": {
+        "serverResource": {
+          "$ref": "#/definitions/Server",
+          "readOnly": true,
+          "description": "Represents resource properties of target server"
+        },
+        "serverParameters": {
```

Comment:
>serverParameters [](http://example.com/codeflow?start=9&length=16)

serverConfiguration, or just configuration?

Reply:
ServerConfiguration and ServerParameters means the same thing. Wanted the name to be more user friendly and readable so named it as ServerParameters.

Reply:
That seems more confusing than friendly: the resource type is named "configurations" in the API. And seeing this serverParameters property, I'd also expect firewallRules to be named serverFirewallRules, for example. Since object is named "targetServerDetails", I think it sets the context as a server already. This approach seems clearer to me:

"targetServerDetails": {
  "resource": { ... }
  "configuration": { ... }
  "firewallRules": [ ... ]
}

WDYT?

Reply:
Changed it to
"targetServerDetails": {
"resource": { ... }
"configurations": { ... }
"firewallRules": [ ... ]
}